infrastructure/terraform/modules/gcp-gke/main.tf

```diff
@@ -0,0 +1,77 @@
+provider "google" {
+  project = var.project_id
+  region  = var.region
+}
+
+# VPC
+resource "google_compute_network" "vpc" {
+  name                    = var.vpc_name
+  auto_create_subnetworks = "false"
+}
+
+# Subnet
+resource "google_compute_subnetwork" "subnet" {
+  name          = "${var.project_id}-subnet-public-0"
+  region        = var.region
+  network       = google_compute_network.vpc.name
+  ip_cidr_range = var.public_subnets[0]
+}
+
+# GKE cluster
+resource "google_container_cluster" "gke_core" {
+  name     = "${var.project_id}-gke"
+  location = var.zone
+  
+  remove_default_node_pool = true
+  initial_node_count       = 1
+
+  network    = google_compute_network.vpc.name
+  subnetwork = google_compute_subnetwork.subnet.name
```

Comment:
This parameter is optional I think, so try without it. That way we can also remove the subnetwork resource and the public ip range.

Reply:
We can't create a GCP cluster with no node pool defined, but we want to only use separately managed node pools. So we create the smallest possible default node pool and immediately delete it.

Reply:
And the subnetwork there deleted